diff -Naur src/org/herac/tuxguitar/gui/editors/tab/edit/EditorKit.java src-patch/org/herac/tuxguitar/gui/editors/tab/edit/EditorKit.java
--- src/org/herac/tuxguitar/gui/editors/tab/edit/EditorKit.java	2008-03-24 13:10:08.000000000 -0300
+++ src-patch/org/herac/tuxguitar/gui/editors/tab/edit/EditorKit.java	2008-03-25 10:42:09.000000000 -0300
@@ -2,8 +2,7 @@
 
 import java.util.Iterator;
 
-import org.eclipse.swt.events.MenuDetectEvent;
-import org.eclipse.swt.events.MenuDetectListener;
+
 import org.eclipse.swt.events.MenuEvent;
 import org.eclipse.swt.events.MenuListener;
 import org.eclipse.swt.events.MouseEvent;
@@ -22,7 +21,7 @@
 import org.herac.tuxguitar.song.models.TGBeat;
 import org.herac.tuxguitar.song.models.TGString;
 
-public class EditorKit implements MouseListener,MouseMoveListener,MouseTrackListener,MenuDetectListener,MenuListener{
+public class EditorKit implements MouseListener,MouseMoveListener,MouseTrackListener,MenuListener{
 	
 	public static final int MOUSE_MODE_SELECTION = 1;
 	public static final int MOUSE_MODE_EDITION = 2;
@@ -42,7 +41,6 @@
 		this.tablature.addMouseListener(this);
 		this.tablature.addMouseMoveListener(this);
         this.tablature.addMouseTrackListener(this);
-        this.tablature.addMenuDetectListener(this);
         this.setDefaults();
 	}
 
@@ -195,10 +193,6 @@
 		}
 	}
 
-	public void menuDetected(MenuDetectEvent e) {
-		this.menuOpen = true;
-	}
-
 	public void menuShown(MenuEvent e) {
 		this.menuOpen = true;
 		this.select(this.menu.x, this.menu.y);
